elasticsearch/run.sh

```diff
@@ -60,4 +69,72 @@ else
 	exit 1
 fi
 
+
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f ${LOG_FILE}
+	echo -n "Checking if Elasticsearch is ready on ${ES_REST_BASEURL} "
+	while ! curl -s -k \
+			--cert ${secret_dir}admin-cert \
+			--key  ${secret_dir}admin-key \
+			--max-time ${max_time} \
+			-o ${LOG_FILE} \
+			${ES_REST_BASEURL} && [ ${timeouted} == false ]
+	do
+		echo -n "."
+		sleep ${RETRY_INTERVAL}
+		(( retry -= 1 ))
+		if (( retry == 0 )) ; then
+			timeouted=true
+		fi
+	done
+
+	# Test for response code 200 in Elasticsearch output. This can be sensitive to Elasticsearch version.
+	if [ -f ${LOG_FILE} ] && grep -q "200" ${LOG_FILE} ; then
+		echo "- connection successful"
+	else
+		if [ ${timeouted} == true ] ; then
+			echo -n "[timeout] "
+		fi
+		echo "failed"
+		cat ${LOG_FILE}
+		exit 1
+	fi
+}
+
+verify_or_add_index_templates() {
+	wait_for_port_open
+	# Try to wait for cluster become more stable before index template being pushed in.
+	# Give up on timeout and continue...
+	curl -v -s -k -X GET \
+		--cert ${secret_dir}admin-cert \
+		--key  ${secret_dir}admin-key \
+		"${ES_REST_BASEURL}/_cluster/health?wait_for_status=yellow&timeout=${max_time}s"
+
+	shopt -s failglob
+	for template_file in /usr/share/elasticsearch/index_templates/*.json
+	do
+		template=`basename $template_file`
```

Comment:
Index templates versioning and dealing with this is whole a different debate. We definitely want to address this in the future but I am not sure we have a clear strategy what we want to do about this. Things can be more complicated because clients may want to introduce their custom index templates or modify our templates...

@t0ffel @portante feel free to jump in if we already know, otherwise we can probably leave it as it is now and get back to this later.